File: miniclaw/bus/message.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import Any


@dataclass
class InboundMessage:
    channel: str
    sender_id: str
    chat_id: str
    content: str
    timestamp: datetime = field(default_factory=datetime.now)
    metadata: dict[str, Any] = field(default_factory=dict)


@dataclass
class OutboundMessage:
    channel: str
    chat_id: str
    content: str
    reply_to: str | None = None
    metadata: dict[str, Any] = field(default_factory=dict)

File: src/bus/message.ts
export type MessageMetadataValue = string | number | boolean | null | undefined;
export type MessageMetadata = Record<string, MessageMetadataValue>;

export interface InboundMessage {
	channel: string;
	sender_id: string;
	chat_id: string;
	content: string;
	timestamp?: Date;
	metadata?: MessageMetadata;
}

/** Outbound message sent to external systems */
export interface OutboundMessage {
	channel: string;
	chat_id: string;
	content: string;
	reply_to?: string;
	metadata?: MessageMetadata;
}
